benchmarl/environments/magent/common.py

```diff
@@ -0,0 +1,126 @@
+#  Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+#  This source code is licensed under the license found in the
+#  LICENSE file in the root directory of this source tree.
+#
+
+from typing import Callable, Dict, List, Optional
+
+from torchrl.data import CompositeSpec
+from torchrl.envs import EnvBase, PettingZooWrapper
+
+from magent2.environments import (
+    adversarial_pursuit_v4,
+    battle_v4,
+    battlefield_v5,
+    combined_arms_v6,
+    gather_v5,
+    tiger_deer_v4
+)
+
+from benchmarl.environments.common import Task
+
+from benchmarl.utils import DEVICE_TYPING
+
+
+class MAgentTask(Task):
+    """Enum for MAgent2 tasks."""
+
+    ADVERSARIAL_PURSUIT = None
+    BATTLE = None
+    BATTLEFIELD = None
+    COMBINED_ARMS = None
+    GATHER = None
+    TIGER_DEER = None
+
+    def get_env_fun(
+            self,
+            num_envs: int,
+            continuous_actions: bool,
+            seed: Optional[int],
+            device: DEVICE_TYPING,
+    ) -> Callable[[], EnvBase]:
+
+        return lambda: PettingZooWrapper(
+            env=self.__get_env(),
+            return_state=True,
+            seed=seed,
+            done_on_any=False,
+            use_mask=True,
```

Comment:
Benchmarl does not support reading the mask for dead agents, can you explain why we set it true?

This is normally true when you have variable number of agents.

In case you just have done agents this can be false

Reply:
Sorry @matteobettini , but I had reached a dead end at this point. At first I thought the number of agents would not change, but now I'm sure it will. If I don't set `use_mask`, this error is thrown because the number of agents changed:

```python
ValueError: Dead agents found in the environment, you need to set use_mask=True to allow this.
```

So, merging won't be possible until BenchMARL supports read mask. However, I've been testing some benchmarks with [adversarial pursuit](https://magent2.farama.org/environments/adversarial_pursuit/) that don't have dead agents and they all work fine. So this is the only issue.

Reply:
I will continue working on fine-tuning all possible algorithms for `adversarial_pursuit` since it is mandatory for my thesis, and I will also try to add an algorithm based on graph neural networks. In the meantime, I will be keeping an eye on the development of this library and if you want help with this problem or others, do not hesitate to ask me.

Reply:
I see, that is unfortunate. I really don't understand this thing that PettingZoo does of removing agents when they are done. This makes everything more complictated as all data now have a variable shape.

> I will continue working on fine-tuning all possible algorithms for adversarial_pursuit since it is mandatory for my thesis, and I will also try to add an algorithm based on graph neural networks. In the meantime, I will be keeping an eye on the development of this library and if you want help with this problem or others, do not hesitate to ask me.

Cool! Maybe we can add that task alone for now?

Aloso just FYI we have a gnn model that can be used with all the algos in case this helps.

Reply:
Ok, I deleted the other tasks. What other task should I add before I mark it as ready for review?